### PR TITLE
Prefer local images in all migration steps (bsc#1227244)

### PR DESCRIPTION
**Pull request by rjmateus**
## What does this PR change?

Fix: https://bugzilla.suse.com/show_bug.cgi?id=1227244

The migration process has several different steps that need to use images. The image path, and if it should be local or remote was done in the first to steps but not on the next ones. Move the server image computation to outside, and passed that as an argument.

## Test coverage
- No tests: **add explanation**
- No tests: already covered
- Unit tests were added

- [ ] **DONE**

## Links

Issue(s): #

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: mgradm/shared/podman/podman.go [hunk 1]
```diff
@@ -199,25 +199,8 @@ func RunMigration(serverImage string, pullPolicy string, sshAuthSocket string, s
...
-	}
-
-	preparedImage, err := podman.PrepareImage(serverImage, pullPolicy, pullArgs...)
-	if err != nil {
-		return "", "", "", err
-	}
-
 	log.Info().Msg(L("Migrating server"))
-	if err := podman.RunContainer("uyuni-migration", preparedImage, utils.ServerVolumeMounts, extraArgs,
+	if err := podman.RunContainer("uyuni-migration", serverImage, utils.ServerVolumeMounts, extraArgs,
```
Comment (aaannz) [inline mgradm/shared/podman/podman.go:203]: pullPolicy is not used now, is it? So rather then change `preparedImage` -> `serverImage`, how about changing method arguments to `preparedImage` and removign `pullPolicy` option?

File: mgradm/cmd/migrate/podman/utils.go [hunk 2]
```diff
@@ -31,11 +31,29 @@ func migrateToPodman(globalFlags *types.GlobalFlags, flags *podmanMigrateFlags,
 		return utils.Errorf(err, L("cannot compute image"))
 	}
 
+	// FIXME all this code should be centralized. Now it being called in several different places.
```
Comment (aaannz) [inline mgradm/cmd/migrate/podman/utils.go:34]: We have this centralized in one of our million of shared dirs - https://github.com/uyuni-project/uyuni-tools/blob/main/shared/podman/utils.go#L176

So should be extracted there.

And no, I don't know why we have 2 Inspect methods :/

**Review by aaannz — APPROVED** [review]
LGTM

One nitpick about args naming.

File: mgradm/cmd/migrate/podman/utils.go (hunk 2 above)
Comment (rjmateus) [reply mgradm/cmd/migrate/podman/utils.go:34]: sure, but in the expect, it doesn't return the `preparedImage`, and that is one of the thing we need, and what should be used to deploy. 
We where calling the prepare image name in several places, and missing in others

File: mgradm/shared/podman/podman.go (hunk 1 above)
Comment (rjmateus) [reply mgradm/shared/podman/podman.go]: Good catch, I have change it according

**Pull request merged**